### PR TITLE
(DOCSP-26638): Realm React Guidance Epic

### PR DESCRIPTION
**Pull request by mongomoe**
### Jira (Epic)

- https://jira.mongodb.org/browse/DOCSP-23395

### Staged Changes

- [React Native SDK Docs (changes throughout)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/realm-react-guidance/sdk/react-native/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [N/A] Create Jira ticket for corresponding docs-app-services update(s), if any
- [N/A] Checked/updated Admin API
- [N/A] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Review by takameyer — COMMENTED** [review]
Review so far.

**Review by mongodben — APPROVED** [review]
lgtm. only thing that needs some updating is some line highlighting.

per discussion, will do as fast follow

File: source/sdk/react-native/realm-database/change-an-object-model.txt [hunk 1]
```diff
@@ -50,31 +51,30 @@ A **schema version** identifies the state of a :ref:`realm schema
...
 .. important:: Increment Versions Monotonically
    
-   :ref:`Migrations <react-native-migrations>` must update a realm to a
-   higher schema version. Realm Database will throw an error if a client
-   application opens a realm with a schema version that is lower than
-   the realm's current version or if the specified schema version is the
-   same as the realm's current version but includes different
-   :ref:`object schemas <react-native-object-schemas>`.
+   :ref:`Migrations <react-native-migrations>` must update a realm to a higher 
+   schema version. Realm Database throws an error if a client application
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply source/sdk/react-native/realm-database/change-an-object-model.txt:61]: Good point! "Realm Database" is an older term docs used to distinguish the mobile database from the backend as a service when we had Realm and MongoDB Realm.

We should change instances of "Realm Database" to just "Realm". I'll make sure it happens in the RN docs upgrade epic, where we're focusing more on wording.

File: examples/react-native/__tests__/js/realm-database/change-an-object-model-test.jsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
+import Realm from 'realm';
+import {createRealmContext} from '@realm/react';
+
+describe('Change an Object Model Tests', () => {
```
[the remark this answers is not in the document]
Comment (krollins-mdb) [reply examples/react-native/__tests__/js/realm-database/change-an-object-model-test.jsx:4]: Addressed this in a new PR: https://github.com/mongodb/docs-realm/pull/2692

**Pull request merged**